packages/proxy/lib/http/util/csp-header.ts

```diff
@@ -0,0 +1,74 @@
+import type { OutgoingHttpHeaders } from 'http'
+
+const cspRegExp = /[; ]*([^\n\r; ]+) ?([^\n\r;]+)*/g
+
+export const nonceDirectives = ['script-src-elem', 'script-src', 'default-src']
+
+export const unsupportedCSPDirectives = [
+  /**
+   * In order for Cypress to run content in an iframe, we must remove the `frame-ancestors` directive
+   * from the CSP header. This is because this directive behaves like the `X-Frame-Options='deny'` header
+   * and prevents the iframe content from being loaded if it detects that it is not being loaded in the
+   * top-level frame.
+   */
+  'frame-ancestors',
```

Comment:
I think that yes, in the future this could be modified to not strip entire directives, but directive:value combinations only. The reason I would be hesitant is that CSP is inclusive by nature. Having a policy that says:
```
Content-Security-Policy: frame-ancestors 'self' 'https://fakedomain.com'
```

...would mean we allow iframes from same origin, and `https://fakedomain.com`. So excluding `self` would mean that we **only** allow iframe from the `https://fakedomain.com` origin.

What we MIGHT be able to do is, instead of removing the directive, augment the `frame-ancestors` directive and supply it with the origin of the Cypress run...the same way we are supplying the `nonce` to the script source directives.